code/modules/atmospherics/environmental/LINDA_system.dm

```diff
@@ -53,29 +53,29 @@
 /atom/movable/proc/block_superconductivity() // objects that block air and don't let superconductivity act
 	return FALSE
 
+/**
+ * Immediately recalculates and updates what turfs can share air with this turf.
+ */
 /turf/proc/immediate_calculate_adjacent_turfs()
-	//Basic optimization, if we can't share why bother asking other people ya feel?
 	var/canpass = CANATMOSPASS(src, src, FALSE)
+	var/list/turf/new_atmos_adjacent_turfs = list()
+	var/list/turf/old_atmos_adjacent_turfs = src.atmos_adjacent_turfs
+	var/list/turf/atmos_adjacency_diffs = list()
```

Comment:
why does this list need to exist? just in general why do you need to rework this logic?